src/main/java/com/eprosima/fastcdr/idl/templates/TypesSourcev1.stg

```diff
@@ -0,0 +1,2469 @@
+$bitset.name$::$bitset.name$(
+        const $bitset.name$& x)
+    $if(bitset.inheritance)$
+    : $bitset_inherit_copy_init(bitset.inheritance)$
+    $endif$
+{
+    m_bitset = x.m_bitset;
+}
+
+$bitset.name$::$bitset.name$(
+        $bitset.name$&& x) noexcept
+    $if(bitset.inheritance)$
+    : $bitset_inherit_move_init(bitset.inheritance)$
+    $endif$
+{
+    m_bitset = x.m_bitset;
+}
+
+$bitset.name$& $bitset.name$::operator =(
+        const $bitset.name$& x)
+{
+    $if(bitset.inheritance)$
+        $bitset.inheritance.scopedname$::operator =(x);
+    $endif$
+
+    m_bitset = x.m_bitset;
+
+    return *this;
+}
+
+$bitset.name$& $bitset.name$::operator =(
+        $bitset.name$&& x) noexcept
+{
+    $if(bitset.inheritance)$
+        $bitset.inheritance.scopedname$::operator =(std::move(x));
+    $endif$
+
+    m_bitset = x.m_bitset;
+
+    return *this;
+}
+
+bool $bitset.name$::operator ==(
+        const $bitset.name$& x) const
+{
+    $if(bitset.inheritance)$
+        if ($bitset.inheritance.scopedname$::operator !=(x)) return false;
+    $endif$
+
+    return m_bitset == x.m_bitset;
+}
+
+bool $bitset.name$::operator !=(
+        const $bitset.name$& x) const
+{
+    return !(*this == x);
+}
+
+$if(ctx.anyCdr)$
+size_t $bitset.name$::getCdrSerializedSize(
+        const $bitset.name$&,
+        size_t current_alignment)
+{
+    size_t initial_alignment = current_alignment;
+
+    current_alignment += sizeof(helper::FindType<$bitset.fullBitSize$>::type)$if(ctx.cdr)$ +
+        eprosima::fastcdr::Cdr::alignment(current_alignment, sizeof(helper::FindType<$bitset.fullBitSize$>::type))$endif$;
+
+    return current_alignment - initial_alignment;
+}
+
+$endif$
+
+$if(ctx.cdr)$
+void $bitset.name$::serialize(
+        eprosima::fastcdr::Cdr& scdr) const
+{
+    $if(!bitset.scope.empty)$
+    using namespace $bitset.scope$;
+    $endif$
+
+    helper::FindType<$bitset.fullBitSize$>::type bits = static_cast<helper::FindType<$bitset.fullBitSize$>::type>(bitset().to_ullong());
+    scdr << bits;
+}
+
+void $bitset.name$::deserialize(
+        eprosima::fastcdr::Cdr& dcdr)
+{
+    $if(!bitset.scope.empty)$
+    using namespace $bitset.scope$;
+    $endif$
+
+    helper::FindType<$bitset.fullBitSize$>::type bits;
+    dcdr \>> bits;
+    bitset(std::bitset<$bitset.fullBitSize$>(bits));
+}
+
+$endif$
+
+$if(ctx.fastcdr)$
+void $bitset.name$::serialize(
+        eprosima::fastcdr::FastCdr& scdr) const
+{
+    $if(!bitset.scope.empty)$
+    using namespace $bitset.scope$;
+    $endif$
+
+    helper::FindType<$bitset.fullBitSize$>::type bits = static_cast<helper::FindType<$bitset.fullBitSize$>::type>(bitset().to_ullong());
+    scdr << bits;
+}
+
+void $bitset.name$::deserialize(
+        eprosima::fastcdr::FastCdr& dcdr)
+{
+    $if(!bitset.scope.empty)$
+    using namespace $bitset.scope$;
+    $endif$
+
+    helper::FindType<$bitset.fullBitSize$>::type bits;
+    dcdr \>> bits;
+    bitset(std::bitset<$bitset.fullBitSize$>(bits));
+}
+
+$endif$
```

Comment:
This code can be unified without distinguishing between cdr|fastcdr

Reply:
The goal was to use the old STG, not improve it, doesn't it?